test/algorithms/KirlikSayin.jl

```diff
@@ -0,0 +1,407 @@
+  MOI.set(model, MOI.Silent(), true)
+  x = MOI.add_variables(model, n)
+  MOI.add_constraint.(model, x, MOI.ZeroOne())
+  MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+  MOI.add_constraint(model, sum(1.0 * x[i] for i in 1:n), MOI.LessThan(W))
+  f = MOI.Utilities.operate(
+      vcat,
+      Float64,
+      [sum(1.0 * C[i, j] * x[j] for j in 1:n) for i in 1:size(C, 1)]...,
+  )
+  MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
+  MOI.optimize!(model)
+  X_E = Float64[
+    0	1	0	0	0	1	0	0	0	0	0	0	0	1	0	0	0	1	0	0	0	0	0	0	1
+    0	0	1	0	0	0	1	0	0	0	0	0	0	1	0	1	0	0	0	0	0	0	0	0	1
+    0	1	0	0	0	0	0	1	0	0	0	0	0	1	0	1	0	0	0	0	0	0	0	0	1
+    0	0	1	0	0	1	0	0	0	0	0	0	0	1	0	0	1	0	0	0	0	0	0	0	1
+    0	0	1	0	0	0	0	0	1	0	0	1	0	0	0	1	0	0	0	0	0	0	0	0	1
+    0	0	1	0	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	1	0	0	0	1	0
+    0	0	1	0	0	1	0	0	0	0	0	1	0	0	0	0	0	0	1	0	0	0	0	0	1
+    0	0	0	0	1	1	0	0	0	0	0	1	0	0	0	0	0	1	0	0	0	0	0	1	0
+    0	1	0	0	0	1	0	0	0	0	0	0	1	0	0	0	0	0	1	0	0	0	0	0	1
+    0	1	0	0	0	1	0	0	0	0	0	0	0	1	0	0	0	0	0	1	0	0	1	0	0
+    0	1	0	0	0	1	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	0	0	1	0
+    0	1	0	0	0	0	0	0	1	0	1	0	0	0	0	0	0	1	0	0	0	0	0	0	1
+    0	0	1	0	0	0	0	0	0	1	0	1	0	0	0	1	0	0	0	0	0	0	0	1	0
+    0	0	0	0	1	0	0	0	1	0	0	1	0	0	0	1	0	0	0	0	0	0	1	0	0
+    0	0	1	0	0	0	1	0	0	0	1	0	0	0	0	0	0	0	0	1	0	0	0	1	0
+    1	0	0	0	0	0	0	0	1	0	0	1	0	0	0	0	0	0	0	1	0	0	1	0	0
+    0	1	0	0	0	0	0	0	1	0	1	0	0	0	0	0	0	0	0	1	0	0	1	0	0
+    0	1	0	0	0	0	0	0	0	1	1	0	0	0	0	0	0	1	0	0	0	0	0	1	0
+    0	1	0	0	0	0	0	1	0	0	1	0	0	0	0	0	0	0	0	1	0	0	0	1	0
+    0	1	0	0	0	0	0	0	0	1	0	0	0	1	0	1	0	0	0	0	0	0	1	0	0
+    0	1	0	0	0	0	0	0	0	1	1	0	0	0	0	0	0	0	1	0	0	0	1	0	0
+  ]
+  Y_N = Float64[
+    16	61	47
+    17	43	71
+    18	47	67
+    20	52	54
+    22	37	63
+    23	43	44
+    22	54	47
+    28	66	39
+    34	60	42
+    24	39	45
+    35	49	39
+    37	55	36
+    28	33	58
+    35	38	56
+    39	43	41
+    38	33	53
+    45	33	34
+    43	51	31
+    40	47	37
+    29	29	59
+    50	40	32
+  ]
+  N = MOI.get(model, MOI.ResultCount())
+  x_sol = hcat([MOI.get(model, MOI.VariablePrimal(i), x) for i in 1:N]...)
+  @test isapprox(x_sol, X_E'; atol = 1e-6)
+  y_sol = hcat([MOI.get(model, MOI.ObjectiveValue(i), x) for i in 1:N]...)
+  @test isapprox(y_sol, Y_N'; atol = 1e-6)
+  return
+end
+
+function test_assignment_max_p3()
+  p = 3
+  n = 5
+  C = [
+    6 1 20 2 3
+    2 6 9 10 18
+    1 6 20 5 9
+    6 8 6 9 6
+    7 10 10 6 2
+    17 20 8 8 20
+    10 13 1 10 15
+    4 11 1 13 1
+    19 13 7 18 17
+    15 3 5 1 11
+    10 7 1 19 12
+    2 15 12 10 3
+    11 20 16 12 9
+    10 15 20 11 7
+    1 9 20 7 6
+  ]
+  C = permutedims(reshape(C, (n, p, n)), [2, 1, 3])
+  model = MOA.Optimizer(HiGHS.Optimizer)
+  MOI.set(model, MOA.Algorithm(), MOA.KirlikSayin())
+  MOI.set(model, MOI.Silent(), true)
+  x = [MOI.add_variable(model) for i = 1:n, j= 1:n]
+  MOI.add_constraint.(model, x, MOI.ZeroOne())
+  for i in 1:n
+    MOI.add_constraint(model, sum(x[i, :]), MOI.EqualTo(1))
```

Comment:
```suggestion
    MOI.add_constraint(model, sum(1.0 * x[i, j] for j in 1:n), MOI.EqualTo(1))
```
We need to be explicit about the coefficient type here. You can't call `sum(::Vector{MOI.VariableIndex})` in MOI.